src/connectors/crowdstrike_devices.py

```diff
@@ -0,0 +1,279 @@
+def get_token_basic(client_id: str, client_secret: str) -> str:
+    headers: dict = {
+        "Content-Type": "application/x-www-form-urlencoded;charset=UTF-8"}
+    try:
+        log.debug(f"Preparing POST: url={CROWDSTRIKE_AUTH_TOKEN_URL}")
+        req: dict = requests.post(
+            CROWDSTRIKE_AUTH_TOKEN_URL,
+            headers=headers,
+            auth=requests.auth.HTTPBasicAuth(client_id, client_secret),
+        )
+        req.raise_for_status()
+    except requests.HTTPError as http_err:
+        log.error(f"Error GET: url={CROWDSTRIKE_AUTH_TOKEN_URL}")
+        log.error(f"HTTP error occurred: {http_err}")
+        raise http_err
+    try:
+        credential = req.json()
+    except Exception as json_error:
+        log.debug(f"JSON error occurred: {json_error}")
+        log.debug(f"requests response {req}")
+        raise (json_error)
+    try:
+        access_token = credential["access_token"]
+    except BaseException:
+        log.error("error auth request token")
+        raise AttributeError("error auth request token")
+    return access_token
+
+
+# Parse out the offset value from the result.
+def get_offset_from_devices_results(result: dict) -> str:
+    if not isinstance(result, dict):
+        log.error("the result is not a dict")
+        raise TypeError("the result from Crowdstrike is not a dict")
+    try:
+        offset: str = result["meta"]["pagination"]["offset"]
+    except BaseException:
+        log.error("the offset is not present")
+        log.error(result.get("meta", "No meta"))
+        raise AttributeError("the offset is not present inside meta")
+    return offset
+
+
+# Perform a generic API call
+def get_data(token: str, url: str, params: dict = {}) -> dict:
+    headers: dict = {"Authorization": f"Bearer {token}"}
+    try:
+        log.debug(f"Preparing GET: url={url} with params={params}")
+        req = requests.get(url, params=params, headers=headers)
+        req.raise_for_status()
+    except HTTPError as http_err:
+        log.error(f"Error GET: url={url}")
+        log.error(f"Error GET: url={url}")
+        log.error(f"HTTP error occurred: {http_err}")
+        raise http_err
+    try:
+        json = req.json()
+
+    except Exception as json_error:
+        log.debug(f"JSON error occurred: {json_error}")
+        log.debug(f"requests response {req}")
+        json = {}
+    return json
+
+
+# Helper function to format the /devices endpoint parameters
+def create_url_params_get_devices(url: str, resources: list) -> str:
+    params_get_devices: str = "?"
+    for id in resources:
+        params_get_devices += "&ids=" + id
+    return url + params_get_devices
+
+
+def connect(connection_name, options):
+    table_name = f'crowdstrike_devices_{connection_name}_connection'
+    landing_table = f'data.{table_name}'
+
+    client_id = options['client_id']
+    client_secret = options['client_secret']
+
+    comment = yaml_dump(
+        module='crowdstrike_devices')
```

Comment:
I think this needs to be something like 

```
comment = yaml_dump(module='crowdstrike_devices', **options)
```

Otherwise, you're not passing the client_id and client_secret back to the ingest method when you call it; the options are parsed from the comment when the connector calls ingest().